onnxruntime/python/tools/transformers/convert_beam_search.py

```diff
@@ -0,0 +1,459 @@
+        sentences = ["The product is released", "I enjoy walking in the park", "Test best way to invest"]
+
+    inputs = tokenizer(sentences, return_tensors='pt', padding=True)
+    input_ids = inputs["input_ids"]
+    attention_mask = inputs["attention_mask"]
+
+    bad_words = "walk in park"
+    bad_words_ids = tokenizer.encode(bad_words, add_prefix_space=True)
+    bad_words_ids = [[word_id] for word_id in bad_words_ids]  # Convert to list of list
+    if use_vocab_mask:
+        print("bad_words_ids", bad_words_ids)
+    else:
+        bad_words_ids = None
+
+    global config
+    config = model.config
+    eos_token_id = config.eos_token_id
+    pad_token_id = config.eos_token_id
+    vocab_size = config.vocab_size
+
+    torch_decoded_sequences = []
+    if not args.disable_parity:
+        print('-' * 50)
+        print("Test PyTorch model and beam search with huggingface transformers...")
+        beam_outputs = model.generate(input_ids=input_ids,
+                                      attention_mask=attention_mask,
+                                      max_length=args.max_length,
+                                      min_length=args.min_length,
+                                      num_beams=args.num_beams,
+                                      early_stopping=args.early_stopping,
+                                      no_repeat_ngram_size=args.no_repeat_ngram_size,
+                                      eos_token_id=eos_token_id,
+                                      pad_token_id=pad_token_id,
+                                      num_return_sequences=args.num_return_sequences,
+                                      temperature=args.temperature,
+                                      length_penalty=args.length_penalty,
+                                      repetition_penalty=args.repetition_penalty,
+                                      bad_words_ids=bad_words_ids,
+                                      return_dict_in_generate=True,
+                                      output_scores=True)
+        print("input_ids", input_ids)
+        print("huggingface transformers outputs:")
+        print("sequences", beam_outputs.sequences)
+        if args.output_sequences_scores:
+            print("sequences_scores", beam_outputs.sequences_scores)
+        if args.output_token_scores:
+            print("scores", beam_outputs.scores)
+        for i, sequence in enumerate(beam_outputs.sequences):
+            decoded_sequence = tokenizer.decode(sequence, skip_special_tokens=True)
+            torch_decoded_sequences.append(decoded_sequence)
+            print("{}: {}".format(i, decoded_sequence))
+
+    print('-' * 50)
+    print("Test ONNX model and bream search with onnxruntime...")
+
+    ort_session = create_ort_session(args.output, args.use_gpu)
+
+    vocab_mask = np.ones((vocab_size), dtype=np.int32)
+    if use_vocab_mask:
+        for bad_word_id in bad_words_ids:
+            vocab_mask[bad_word_id] = 0
+
+    inputs = {
```

Comment:
the arguments passed to conversion script are used here to test. However, the graph itself would only have defaults since these are inputs. We can remove these optional values to parameters

[--temperature TEMPERATURE]
[--length_penalty LENGTH_PENALTY] 
[--repetition_penalty REPETITION_PENALTY]

unless I am missing something.

Reply:
To avoid confusion, how about using initializers to store these parameters in ONNX graph?

Reply:
why does convert_beam_search.py need these parameters passed in?

Reply:
This script generates ONNX model, and it also serves an example of how to use ONNX model.

So we exposed the parameters so that user could try different values to see its impact. Currently, those parameters are used in inference only, and not store in ONNX model. In the future, we could add an option that user to choose which parameters needed to be store as initializers in ONNX model so that they need not specify in inputs for inference.